src/Template/Element/sort_option.ctp

```diff
@@ -0,0 +1,17 @@
+<md-menu-item>
+    <md-button>
+        <md-icon>
+            <?php 
+                if (strcmp($this->Paginator->sortKey(),$param) == 0 && strcmp($this->Paginator->sortDir(),$direction) == 0) {
```

Comment:
Why do you use `strcmp` instead of a "simple" string comparison `s1 == s2`?

Reply:
Defensive programming. In another project I was working I had many issues because of string comparison as some inputs where not strings and this prevent many errors. So I do like this because I am used to.

Reply:
Anyway I will change this while taking care of the other changes listed below.

Reply:
If you want to be sure that both operands are strings, you should use `===`, i.e. strict comparison. (Actually we should probably use always `===` when testing whether two strings are equal, see https://stackoverflow.com/questions/3333353/string-comparison-using-vs-strcmp)
I think using `strcmp` is only useful if you want to know which string is greater than the other.

Reply:
Ok I changed to `===`